packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/FLTGoogleMapJSONConversions.m

```diff
@@ -141,4 +141,28 @@ + (nullable GMSCameraUpdate *)cameraUpdateFromChannelValue:(NSArray *)channelVal
   }
   return nil;
 }
+
++ (NSArray<GMSStrokeStyle *> *)strokeStylesFromPatterns:(NSArray<NSArray *> *)patterns
+                                            strokeColor:(UIColor *)strokeColor {
+  NSMutableArray *strokeStyles = [[NSMutableArray alloc] initWithCapacity:[patterns count]];
+  for (unsigned int i = 0; i < [patterns count]; i++) {
+    NSString *patternType = patterns[i][0];
+
+    UIColor* color = [patternType isEqualToString:@"gap"] ? [UIColor clearColor] : strokeColor;
+    strokeStyles[i] = [GMSStrokeStyle solidColor:color];
+  }
+    
+
+  return strokeStyles;
+}
+
++ (NSArray<NSNumber *> *)spanLengthsFromPatterns:(NSArray<NSArray *> *)patterns {
+  NSMutableArray *lengths = [[NSMutableArray alloc] initWithCapacity:[patterns count]];
+  for (unsigned int i = 0; i < [patterns count]; i++) {
```

Comment:
This should use fast enumeration rather than indexing, and `addObject:` to `lengths`.